sympy/polys/domains/tests/test_domains.py

```diff
@@ -1171,6 +1171,11 @@ def test_EX_EXRAW():
     assert EX.unify(EXRAW) == EXRAW
 
 
+def test_EX_ordering():
+    elements = [EX(1), EX(x), EX(3)]
+    assert sorted(elements) == [EX(1), EX(3), EX(x)]
```

Comment:
This makes me a little nervous because it creates something that can be sorted if decoration is `EX` that would fail normally, like `sorted([1,x])`. Can `list(ordered(...))` be used? (That gives the expected output here, BTW.)

Reply:
The existing `factor_list` code depends on being able to order the elements of arbitrary domains:
https://github.com/sympy/sympy/blob/5e9de1c5b073e29fcc30fce78a73b8c0f363fc21/sympy/polys/polytools.py#L6079-L6092
Here `rep` is a list of domain elements which in this case happen to be `EX`. It isn't currently possible to compare `EX` hence the failure:
```python
In [1]: factor_list((x - pi*sqrt(2))*(x + pi*sqrt(2)), x)
---------------------------------------------------------------------------
...
File ~/current/sympy/sympy.git/sympy/polys/polytools.py:6092, in _sorted_factors(factors, method)
   6089         rep = poly.rep.rep
   6090         return (len(rep), len(poly.gens), exp, str(poly.domain), rep)
-> 6092 return sorted(factors, key=key)

TypeError: '<' not supported between instances of 'Expression' and 'Expression'
```
The purpose of the test here is to ensure that the expectation of `factor_list` continues to be upheld.